contentcuration/contentcuration/frontend/shared/utils/validation.js

```diff
@@ -0,0 +1,397 @@
+    [ValidationErrors.TITLE_REQUIRED]: translator.$tr('titleRequired'),
+    [ValidationErrors.LICENSE_REQUIRED]: translator.$tr('licenseRequired'),
+    [ValidationErrors.COPYRIGHT_HOLDER_REQUIRED]: translator.$tr('copyrightHolderRequired'),
+    [ValidationErrors.LICENSE_DESCRIPTION_REQUIRED]: translator.$tr('licenseDescriptionRequired'),
+    [ValidationErrors.MASTERY_MODEL_REQUIRED]: translator.$tr('masteryModelRequired'),
+    [ValidationErrors.MASTERY_MODEL_M_REQUIRED]: translator.$tr('masteryModelMRequired'),
+    [ValidationErrors.MASTERY_MODEL_M_WHOLE_NUMBER]: translator.$tr('masteryModelMWholeNumber'),
+    [ValidationErrors.MASTERY_MODEL_M_GT_ZERO]: translator.$tr('masteryModelMGtZero'),
+    [ValidationErrors.MASTERY_MODEL_M_LTE_N]: translator.$tr('masteryModelMLteN'),
+    [ValidationErrors.MASTERY_MODEL_N_REQUIRED]: translator.$tr('masteryModelNRequired'),
+    [ValidationErrors.MASTERY_MODEL_N_WHOLE_NUMBER]: translator.$tr('masteryModelNWholeNumber'),
+    [ValidationErrors.MASTERY_MODEL_N_GT_ZERO]: translator.$tr('masteryModelNGtZero'),
+  };
+
+  return messages[error];
+}
+
+// Helpers
+export function translateValidator(validator) {
+  return value => (validator(value) === true ? true : _getErrorMsg(validator(value)));
+}
+
+// Validators
+// These functions return an array of validators (validator is
+// a function that returns `true` or an error code)
+// Designed to be compatible with Vuetify's inputs rules
+export function getTitleValidators() {
+  return [value => Boolean(value) || ValidationErrors.TITLE_REQUIRED];
+}
+
+export function getLicenseValidators() {
+  return [value => Boolean(value) || ValidationErrors.LICENSE_REQUIRED];
+}
+
+export function getCopyrightHolderValidators() {
+  return [value => Boolean(value) || ValidationErrors.COPYRIGHT_HOLDER_REQUIRED];
+}
+
+export function getLicenseDescriptionValidators() {
+  return [value => Boolean(value) || ValidationErrors.LICENSE_DESCRIPTION_REQUIRED];
+}
+
+export function getMasteryModelValidators() {
+  return [value => Boolean(value) || ValidationErrors.MASTERY_MODEL_REQUIRED];
+}
+
+export function getMasteryModelMValidators(nValue) {
+  return [
+    value => Boolean(value) || ValidationErrors.MASTERY_MODEL_M_REQUIRED,
+    value => Number.isInteger(Number(value)) || ValidationErrors.MASTERY_MODEL_M_WHOLE_NUMBER,
+    value => value > 0 || ValidationErrors.MASTERY_MODEL_M_GT_ZERO,
+    value => value <= nValue || ValidationErrors.MASTERY_MODEL_M_LTE_N,
+  ];
+}
+
+export function getMasteryModelNValidators() {
+  return [
+    value => Boolean(value) || ValidationErrors.MASTERY_MODEL_N_REQUIRED,
+    value => Number.isInteger(Number(value)) || ValidationErrors.MASTERY_MODEL_N_WHOLE_NUMBER,
+    value => value > 0 || ValidationErrors.MASTERY_MODEL_N_GT_ZERO,
+  ];
+}
+
+// Node validation
+// These functions return an array of error codes
+export function validateNodeTitle(node) {
+  return getTitleValidators()
+    .map(validator => validator(node.title))
```

Comment:
It might be good to use `node.title.trim()` in case someone tries to use a blank title as well

Reply:
I did the same for copyright holder and license description fields.

Reply:
Awesome, thank you!